### PR TITLE
add RemovePerformanceDelegation and change logic for RemoveZoneAndAssociated…

### LINKED ISSUE #757: We need a RemovePerformanceDelegation func to perform deletion.
I see that in this line [here](https://github.com/quicksilver-zone/quicksilver/blob/main/x/interchainstaking/keeper/zones.go#L463) (func RemoveZoneAndAssociatedRecords) cannot be deleted PerformanceDelegation

because when setPerformanceDelegation uses types.GetPerformanceDelegationKey, setDelegation uses types.GetDelegationKey

 So we need a RemovePerformanceDelegation func to perform delegation.
 
 Context: #678

### PR DESCRIPTION
**Pull request by DongLieu**
…Records

## 1. Summary
Fixes #757
context: #678

## 2.Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## 5. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

File: x/interchainstaking/keeper/delegation.go [hunk 1]
```diff
@@ -168,6 +168,15 @@ func (k *Keeper) RemoveDelegation(ctx sdk.Context, chainID string, delegation ty
 	return nil
 }
 
+// RemovePerformanceDelegation removes a performance delegation.
+func (k *Keeper) RemovePerformanceDelegation(ctx sdk.Context, chainID string, delegation types.Delegation) error {
```
Comment (ThanhNhann) [inline x/interchainstaking/keeper/delegation.go:172]: Nice catch 💪 @DongLieu. Can you set up testing for the new function you added, maybe like [this](https://github.com/quicksilver-zone/quicksilver/pull/758/RemovePerformanceDelegation)

Comment (DongLieu) [reply x/interchainstaking/keeper/delegation.go:172]: it's available [here:](https://github.com/quicksilver-zone/quicksilver/pull/678#issuecomment-1786296197) pull #678

Comment (DongLieu) [reply x/interchainstaking/keeper/delegation.go:172]: Ok, I will add it now ,separately.

**Comment by DongLieu** [comment]
@ThanhNhann 
I added test

**Review by ThanhNhann — APPROVED** [review]
Lgtm! Thank @DongLieu

**Pull request merged**